src/scenes/CreateEvent/index.js

```diff
@@ -0,0 +1,652 @@
+              onClosed={this.onSelectResultSetterAddressDialogClosed}
+            />
+          </DialogContent>
+          <DialogActions>
+            <Button color="primary" onClick={onClose}>
+              {intl.formatMessage(messages.cancel)}
+            </Button>
+            <Button type="submit" color="primary" disabled={submitting} variant="raised">
+              {intl.formatMessage(messages.publish)}
+            </Button>
+          </DialogActions>
+        </Form>
+      </Dialog>
+    );
+  }
+}
+
+const mapStateToProps = (state) => ({
+  initialValues: {
+    bettingStartTime: DEFAULT_PICKER_TIME,
+    bettingEndTime: DEFAULT_PICKER_TIME,
+    resultSettingStartTime: DEFAULT_PICKER_TIME,
+    resultSettingEndTime: DEFAULT_PICKER_TIME,
+    outcomes: ['', ''],
+  },
+  txReturn: state.Graphql.get('txReturn'),
+  walletAddresses: state.App.get('walletAddresses'),
+  lastUsedAddress: state.App.get('lastUsedAddress'),
+  syncBlockNum: state.App.get('syncBlockNum'),
+  averageBlockTime: state.App.get('averageBlockTime'),
+});
+
+function mapDispatchToProps(dispatch) {
+  return {
+    createTopicTx: (
+      name,
+      outcomes,
+      resultSetter,
+      bettingStartTime,
+      bettingEndTime,
+      resultSettingStartTime,
+      resultSettingEndTime,
+      creatorAddress,
+    ) => dispatch(graphqlActions.createTopicTx(
+      name,
+      outcomes,
+      resultSetter,
+      bettingStartTime,
+      bettingEndTime,
+      resultSettingStartTime,
+      resultSettingEndTime,
+      creatorAddress,
+    )),
+    getInsightTotals: () => dispatch(appActions.getInsightTotals()),
+    setLastUsedAddress: (address) => dispatch(appActions.setLastUsedAddress(address)),
+    changeFormFieldValue: (field, value) => dispatch(change(FORM_NAME, field, value)),
+  };
+}
+
+const fields = [
+  'name',
+  'outcomes',
+  'resultSetter',
+  'creatorAddress',
+  ID_BETTING_START_TIME,
+  ID_BETTING_END_TIME,
+  ID_RESULT_SETTING_START_TIME,
+  ID_RESULT_SETTING_END_TIME,
+];
+
+const validate = (values, props) => {
+  const { intl } = props;
+
+  const errors = {};
+
+  // check required fields
+  const requiredFields = fields;
+  requiredFields.forEach((field) => {
+    if (!values[field]) {
+      errors[field] = intl.formatMessage(messages.required);
+    }
+  });
+
+  // check dates
+  const bettingStartTime = moment(values.bettingStartTime);
+  const bettingEndTime = moment(values.bettingEndTime);
+  const resultSettingStartTime = moment(values.resultSettingStartTime);
+  const resultSettingEndTime = moment(values.resultSettingEndTime);
+
+  if (bettingStartTime.unix() > bettingEndTime.unix()) {
+    errors.bettingEndTime = intl.formatMessage(messages.validBetEnd);
+  }
+
+  if (bettingEndTime.unix() > resultSettingStartTime.unix()) {
+    errors.resultSettingStartTime = intl.formatMessage(messages.validResultSetStart);
+  }
+
+  if (resultSettingStartTime.unix() > resultSettingEndTime.unix()) {
```

Comment:
same as above. suggest we make a min gap of like 30 mins between start and end times.